src/toolbars/ToolBar.cpp

```diff
@@ -641,6 +646,12 @@ ToolDock *ToolBar::GetDock()
    return dynamic_cast<ToolDock*>(GetParent());
 }
 
+void ToolBar::SetPreferredNeighbors(ToolBarID left, ToolBarID top)
+{
+   mPreferredLeftNeighbor = left;
+   mPreferredTopNeighbor = top;
+}
```

Comment:
You rewrote my proposal with more generality.  Very well.

But as you might guess, I don't really like `DefaultConfigEntry` because it is a failure of open-endedness of the set of toolbars.  I would rather take the time fo make a new Registry that scatters the definition of the default configuration.

Each toolbar should decide where to insert itself.

For that, each toolbar's .cpp file would need to refer to another toolbar, but only the weak for of dependency that mentions a textual id, and if the toolbar module isn't present, there is no harm.

My pull request Open-ended-toolbars does what is needed to eliminate the centralized enumeration and identify toolbars by strings instead.  It doesn't yet go further to build up the default configuration by collecting registry items.